components/supervisor/frontend/src/ide/gitpod-server-compatibility.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { serverUrl } from "../shared/urls";
+
+const versionRegex = new RegExp("^main.(\d+)$")
+
+function getVersionInfo(version: string) {
+    const result = versionRegex.exec(version);
+    if (!result) {
+        return;
+    }
+    return Number(result[1]);
+}
+
+export async function isSaasServerGreaterThan (version: string) {
+    const url = serverUrl.withApi({ pathname: "/version" }).toString();
```

Comment:
Should not we check that we are on gitpod.io?

This check does not makes sense in Self-Hosted.